src/lib/files.ts

```diff
@@ -3,27 +3,33 @@ import { fabric } from "fabric";
 import { HistoryCommand } from "./history";
 import Pages, { PageJSON } from "./pages";
 
-export class AsyncReader extends Promise<FileReader> {
-  constructor(file: File) {
-    if (file instanceof File)
-      super((resolve, reject) => {
-        const reader = new FileReader();
-        reader.onload = () => {
-          resolve(reader);
-        };
-        reader.onerror = reject;
-        reader.readAsText(file);
-      });
-    else super(file);
-  }
+export class AsyncReader {
+  static readAsText = (file: File): Promise<string | ArrayBuffer> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = () => {
+        resolve(reader.result);
+      };
+      reader.onerror = reject;
+      reader.readAsText(file);
+    });
+
+  static readAsDataURL = (file: File): Promise<string | ArrayBuffer> =>
+    new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = () => {
+        resolve(reader.result);
+      };
+      reader.onerror = reject;
+      reader.readAsDataURL(file);
+    });
 }
 
 // manages version compatibility with old document formats
 // change the signature and usages to accommodate new data; this will fill in sample data for missing fields
 export class JSONReader {
-  static async read(reader: AsyncReader): Promise<PageJSON[]> {
-    const json = (await reader).result.toString();
-    const object = JSON.parse(json);
+  static async read(json: Promise<string | ArrayBuffer>): Promise<PageJSON[]> {
```

Comment:
Maybe this should assume a resolved Promise. Just means we need to add awaits to all usages (only in this file)